spec/solver_spec.rb

```diff
@@ -0,0 +1,47 @@
+require_relative '../solver'
+
+describe Solver do
+  context '#solver' do
+    before do
+      @solver = Solver.new
+    end
+
+    describe 'Testing factorial method' do
+      it 'will test the `factorial` of 3 should be 6' do
+        expect(@solver.factorial(3)).to eql(6)
+      end
+
+      it 'will test the `factorial` of 0 should be 0' do
+        expect(@solver.factorial(0)).to eql(0)
+      end
+      #   it 'will test the `factorial` of negative number should raise error' do
+      #     expect {@solver.factorial(1)}.to raise_error(RuntimeError)
+      #   end
```

Comment:
- [ ] Kindly notice that you should add those two test cases, the Negative value which raises an exception as an expectation, and the Special case - when input is 0, the result should equal 1.

```ruby
it 'will test the `factorial` of 0 should be 0' do
    expect(@solver.factorial(0)).to eql(1)
end
it 'will test the `factorial` of negative number should raise error' do
     expect {@solver.factorial(-5)}.to raise_error(RuntimeError)
end
```

Reply:
@ShahierNashaat Thanks. solved